datafusion-cli/Dockerfile

```diff
@@ -15,7 +15,7 @@
 # specific language governing permissions and limitations
 # under the License.
 
-FROM rust:1.80-bookworm AS builder
+FROM rust:1.81-bookworm AS builder
```

Comment:
Per this comment: https://github.com/apache/datafusion/pull/14330#issuecomment-2620005880

maybe we should run the latest stable rust here 🤔

Reply:
I think it makes sense, we do use MSRV to let users with older Rust compiler version to be able to use DF as a crate. However Datafusion-cli is not really part of it and perhaps we can have the latest version yes